config/reader.go

```diff
@@ -33,9 +32,9 @@ var (
 )
 
 var (
-	placeholderRegexp       = regexp.MustCompile(`\$\{[A-Za-z0-9_\.\-]+\}`)
-	verifyPlaceholderRegexp = regexp.MustCompile(`^\$\{([A-Za-z0-9_\.\-]+)\}$`)
-	viamDotDir              = filepath.Join(os.Getenv("HOME"), ".viam")
+	placeholderRegexp = regexp.MustCompile(`\$\{(packages[\.(ml_model|modules)]*\.[A-Za-z0-9_\-]+)\}`)
```

Comment:
I'm not sure that regex is doing what you want. The square brackets denote a character set, not an optional section, so it's basically allowing any number of the the characters `ml_de|ous().`

Also, \w is much shorter for the alphanumeric bit at the end.

```suggestion
	placeholderRegexp = regexp.MustCompile(`\$\{(packages(\.(ml_model|modules))?\.[\w-]+)\}`)
```

Check out https://regex101.com/r/a22G75/1 for a handy tool for testing regex expressions. That link should be to my suggested change.

Reply:
ah, yea i was testing it with optional selection and it worked but you are right this would make it pass for multiple .ml_model/modules paths. Thanks!